src/net/sourceforge/kolmafia/Modifiers.java

```diff
@@ -98,689 +99,15 @@ public class Modifiers {
-    new DoubleModifier(
-        "Clowniness",
-        Pattern.compile("Makes you look (\\d+)% clowny"),
-        Pattern.compile("Clowniness: " + EXPR)),
-    new DoubleModifier(
-        "Maximum PP",
-        Pattern.compile("([+-]\\d+) Max(imum)? Power Point"),
-        Pattern.compile("Maximum PP: " + EXPR)),
-    new DoubleModifier("Plumber Power", Pattern.compile("Plumber Power: " + EXPR)),
-    new DoubleModifier(
-        "Drippy Damage",
-        new Pattern[] {
-          Pattern.compile("([+-]\\d+) Damage vs. creatures of The Drip"),
-          Pattern.compile("([+-]\\d+) Damage against Drip creatures"),
-        },
-        Pattern.compile("Drippy Damage: " + EXPR)),
-    new DoubleModifier("Drippy Resistance", Pattern.compile("Drippy Resistance: " + EXPR)),
-    new DoubleModifier("Energy", Pattern.compile("Energy: " + EXPR)),
-    new DoubleModifier("Scrap", Pattern.compile("Scrap: " + EXPR)),
-    new DoubleModifier("Familiar Action Bonus", Pattern.compile("Familiar Action Bonus: " + EXPR)),
-    new DoubleModifier(
-        "Water",
-        Pattern.compile("Collect (\\d+) water per adventure"),
-        Pattern.compile("Water: " + EXPR)),
-    new DoubleModifier(
-        "Spleen Drop",
-        Pattern.compile("([+-]\\d+)% Spleen Item Drops? [Ff]rom Monsters$"),
-        Pattern.compile("Spleen Drop: " + EXPR)),
-    new DoubleModifier(
-        "Potion Drop",
-        Pattern.compile("([+-]\\d+)% Potion Drops? [Ff]rom Monsters$"),
-        Pattern.compile("Potion Drop: " + EXPR)),
-    new DoubleModifier(
-        "Sauce Spell Damage",
-        new Pattern[] {
-          Pattern.compile("Sauce Spell Damage ([+-]\\d+)$"),
-          Pattern.compile("([+-]\\d+) Sauce Spell Damage"),
-        },
-        Pattern.compile("(?:^|, )Sauce Spell Damage: " + EXPR)),
-    new DoubleModifier(
-        "Monster Level Percent",
-        Pattern.compile("([+-]\\d+)% Monster Level"),
-        Pattern.compile("Monster Level Percent: " + EXPR)),
-    new DoubleModifier("Food Fairy", Pattern.compile("Food Fairy: " + EXPR)),
-    new DoubleModifier("Booze Fairy", Pattern.compile("Booze Fairy: " + EXPR)),
-    new DoubleModifier("Candy Fairy", Pattern.compile("Candy Fairy: " + EXPR)),
-    new DoubleModifier(
-        "Food Fairy Effectiveness", Pattern.compile("Food Fairy Effectiveness: " + EXPR)),
-    new DoubleModifier(
-        "Booze Fairy Effectiveness", Pattern.compile("Booze Fairy Effectiveness: " + EXPR)),
-    new DoubleModifier(
-        "Candy Fairy Effectiveness", Pattern.compile("Candy Fairy Effectiveness: " + EXPR)),
-  };
-
-  public static final int DOUBLE_MODIFIERS = Modifiers.doubleModifiers.length;
-
   private static final HashSet<String> numericModifiers = new HashSet<>();
 
   static {
-    for (int i = 0; i < DOUBLE_MODIFIERS; ++i) {
-      DoubleModifier modifier = Modifiers.doubleModifiers[i];
-      modifierIndicesByName.put(modifier.getName(), i);
+    for (var modifier : DoubleModifier.values()) {
```

Comment:
I will mention that the values() method returns an array of enums.
Java does not have immutable arrays, so that is allocated and filled every time you call that static method.
You might consider making an EnumSet.allOf(DoubleModifer.class) somewhere. Perhaps in Modifiers.java?
Although Java does have immutable Sets, I don't think so for EnumSets, although you could agree not to modify it.
That is likely to be similarly fast to iterate over as an array and should not require memory allocation.

I think you have other iterations of DoubleModifer.values(), so this same comment applies, although I will not repeat it; you can decide what, if anything, you want to do globally.